golang/implode.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"fmt"
+)
+
+func Implode[T string | int](array []T, sep ...string) string {
```

Comment:
Ini bakal limited buat slice of string/int sih. Gimana kalo ada data type lain kaya uint, char, rune, dll ?

urutannya juga kebalik sih, tapi ngga tau gimana caranya bikin optional parameter di depan
kalo pake variadic ga bagus keknya, param di belakang ga bisa dapet type yg beda

Reply:
nah ini yg sempet jadi pertanyaan semalem. soale lihat test case sebelah cuma string dan int aja. ok noted